images/ubuntu2004/scripts/cloudinit_install.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+
+GRUB_CMDLINE_LINUX_DEFAULT="console=tty1 console=ttyS0,115200n8 earlyprintk=ttyS0,115200 rootdelay=300 quiet splash"
+
+echo "ubuntu" | sudo -S apt update
+sudo apt-get -y install cloud-init gdisk netplan.io walinuxagent 
+sudo systemctl stop walinuxagent 
+
+sudo rm -f /etc/cloud/cloud.cfg.d/50-curtin-networking.cfg /etc/cloud/cloud.cfg.d/curtin-preserve-sources.cfg /etc/cloud/cloud.cfg.d/99-installer.cfg /etc/cloud/cloud.cfg.d/subiquity-disable-cloudinit-networking.cfg
+sudo rm -f /etc/cloud/ds-identify.cfg
+sudo rm -f /etc/netplan/*.yaml
+sudo rm -f /etc/cloud/cloud.cfg.d/90_dpkg.cfg
+
+sudo sh -c "cat >> /etc/cloud/cloud.cfg.d/90_dpkg.cfg" << 'EOF'
+datasource_list: [ Azure ]
+EOF
+
+sudo sh -c "cat >> /etc/cloud/cloud.cfg.d/90-azure.cfg" << 'EOF'
+system_info:
+   package_mirrors:
+     - arches: [i386, amd64]
+       failsafe:
+         primary: http://archive.ubuntu.com/ubuntu
+         security: http://security.ubuntu.com/ubuntu
+       search:
+         primary:
+           - http://azure.archive.ubuntu.com/ubuntu/
+         security: []
+     - arches: [armhf, armel, default]
+       failsafe:
+         primary: http://ports.ubuntu.com/ubuntu-ports
+         security: http://ports.ubuntu.com/ubuntu-ports
+EOF
+
+sudo sh -c "cat >> /etc/cloud/cloud.cfg.d/10-azure-kvp.cfg" << 'EOF'
+reporting:
+  logging:
+    type: log
+  telemetry:
+    type: hyperv
+EOF
+
+sudo sed -i 's/Provisioning.Enabled=y/Provisioning.Enabled=n/g' /etc/waagent.conf
+sudo sed -i 's/Provisioning.UseCloudInit=n/Provisioning.UseCloudInit=y/g' /etc/waagent.conf
+sudo sed -i 's/ResourceDisk.Format=y/ResourceDisk.Format=n/g' /etc/waagent.conf
+sudo sed -i 's/ResourceDisk.EnableSwap=y/ResourceDisk.EnableSwap=n/g' /etc/waagent.conf
+
+sudo sh -c "cat >>  /etc/waagent.conf" << 'EOF'
+# For Azure Linux agent version >= 2.2.45, this is the option to configure,
+# enable, or disable the provisioning behavior of the Linux agent.
+# Accepted values are auto (default), waagent, cloud-init, or disabled.
+# A value of auto means that the agent will rely on cloud-init to handle
+# provisioning if it is installed and enabled, which in this case it will.
+Provisioning.Agent=auto
+EOF
+
+sudo cloud-init clean --logs --seed
+sudo rm -rf /var/lib/cloud/
+sudo systemctl stop walinuxagent.service
```

Comment:
It's not clear why we stop in line 7, then again here

Reply:
Thanks, removed the duplicate stop earlier